File: app/titan/model/status.go
// Package model 中集中定义 Titan 全部资源状态枚举（单一真源）。
// 数据库默认值、种子数据与业务代码判断必须使用同一词汇表，
// 禁止在 logic 层散落魔法字符串。
package model

// 流水线执行状态 (titan_pipeline_exec.status / titan_pipeline_step_exec.status)
const (
	ExecStatusPending   = "PENDING"
	ExecStatusRunning   = "RUNNING"
	ExecStatusSuccess   = "SUCCESS"
	ExecStatusFailed    = "FAILED"
	ExecStatusAborted   = "ABORTED"
	ExecStatusSkipped   = "SKIPPED"
	ExecStatusCancelled = "CANCELLED"
)

// IsExecTerminal 判断执行状态是否为终态（终态后不可取消、日志拉取结束）
func IsExecTerminal(status string) bool {
	switch status {
	case ExecStatusSuccess, ExecStatusFailed, ExecStatusAborted, ExecStatusCancelled, ExecStatusSkipped:
		return true
	}
	return false
}

// IsStepTerminal 判断步骤执行状态是否为终态
func IsStepTerminal(status string) bool {
	return IsExecTerminal(status)
}

// 环境-应用绑定状态 (titan_env_app_binding.status)
const (
	BindingStatusPending   = "PENDING"   // 待部署
	BindingStatusDeploying = "DEPLOYING" // 部署中（幂等锁：部署进行中拒绝重复提交）
	BindingStatusRunning   = "RUNNING"   // 运行中
	BindingStatusFailed    = "FAILED"    // 部署失败
	BindingStatusStopped   = "STOPPED"   // 已停止
)

// 制品状态 (titan_artifact.status)
const (
	ArtifactStatusAvailable = "AVAILABLE"
	ArtifactStatusExpired   = "EXPIRED"
)

// 集群状态 (titan_cluster.status)
const (
	ClusterStatusHealthy   = "HEALTHY"
	ClusterStatusUnhealthy = "UNHEALTHY"
	ClusterStatusUnknown   = "UNKNOWN"
)

// 环境状态 (titan_env.status)
const (
	EnvStatusActive   = "ACTIVE"
	EnvStatusInactive = "INACTIVE"
)

// 应用/项目/集成/流水线通用启停状态 (tinyint: 1 启用 0 停用)
const (
	CommonStatusEnabled  = int64(1)
	CommonStatusDisabled = int64(0)
)

File: frontend/apps/admin/src/pages/Titan/status.ts
// 与后端 app/titan/model/status.go 口径对齐的状态枚举（单一真源），
// 前端禁止散落书写状态魔法字符串，统一从此处引用。
// 注意：本文件为 titan 应用 src/constants/status.ts 的同步副本（admin 无法跨应用引用其源码）。
export const EXEC_STATUS = {
  PENDING: 'PENDING',
  RUNNING: 'RUNNING',
  SUCCESS: 'SUCCESS',
  FAILED: 'FAILED',
  ABORTED: 'ABORTED',
  SKIPPED: 'SKIPPED',
  CANCELLED: 'CANCELLED',
} as const;

// 环境-应用绑定状态（含部署幂等锁 DEPLOYING）
export const BINDING_STATUS = {
  PENDING: 'PENDING',
  DEPLOYING: 'DEPLOYING',
  RUNNING: 'RUNNING',
  FAILED: 'FAILED',
  STOPPED: 'STOPPED',
} as const;

export const ARTIFACT_STATUS = {
  AVAILABLE: 'AVAILABLE',
  EXPIRED: 'EXPIRED',
} as const;

export const CLUSTER_STATUS = {
  HEALTHY: 'HEALTHY',
  UNHEALTHY: 'UNHEALTHY',
  UNKNOWN: 'UNKNOWN',
} as const;

export const ENV_STATUS = {
  ACTIVE: 'ACTIVE',
  INACTIVE: 'INACTIVE',
} as const;

// 通用 Tag 快捷配色（与 AntD 预设色对齐）
export const statusTagColor: Record<string, string> = {
  SUCCESS: 'success',
  RUNNING: 'processing',
  FAILED: 'error',
  ABORTED: 'default',
  PENDING: 'default',
};

// 流水线执行状态 → 展示元数据。
// text 用于详情/大盘完整展示，label 用于列表精简展示；
// WAITING_APPROVAL 为人工审批门禁的 UI 过渡态（非 DB 落库枚举）。
export interface ExecStatusMeta {
  color: string;
  text: string;
  label: string;
}

export const execStatusMap: Record<string, ExecStatusMeta> = {
  PENDING: { color: 'default', text: '排队就绪', label: '就绪' },
  RUNNING: { color: 'processing', text: '执行中', label: '执行中' },
  WAITING_APPROVAL: { color: 'warning', text: '等待人工审批', label: '等待审批' },
  SUCCESS: { color: 'success', text: '发布成功', label: '发布成功' },
  FAILED: { color: 'error', text: '执行失败', label: '失败' },
  CANCELLED: { color: 'default', text: '已终止', label: '已终止' },
  ABORTED: { color: 'warning', text: '已中止', label: '已中止' },
  SKIPPED: { color: 'default', text: '已跳过', label: '已跳过' },
};
